### PR TITLE
Parallel page rank

### PR DESCRIPTION
**Pull request by SohamTamba**
Implemented `parallel_pagerank`

nthreads() = 4

`g = loadsnap(:ego_twitter_u)` (|V| = 81306, |E| = 1342310)

`pagerank(g)`: 192.269 ms (10 allocations: 1.86 MiB)

`parallel_pagerank(g)`: 70.239 ms (23 allocations: 3.72 MiB)

`g = random_regular_graph(10000, 2000)` (|V| = 10,000, |E| = 10,000,000)

`pagerank(g):` 1.404 s (10 allocations: 234.75 KiB)

`parallel_pagerank(g)`: 237.902 ms (20 allocations: 469.50 KiB)

*cc* @somil55

### CODE REVIEW

**Comment by SohamTamba** [comment]
The [report](https://codecov.io/gh/JuliaGraphs/LightGraphs.jl/compare/292feb8f367a0e94ce4e18e696eac70dc246caf5...949fd7861502157027e3939b3eb59fbc2584e4f5/diff) shows that `partitioning` is not covered.

Maybe codecov is run when `JULIA_NUM_THREADS` is set to 1?

**Comment by somil55** [comment]
@SohamTamba Nice work! I will review it tomorrow.

**Comment by SohamTamba** [comment]
@sbromberger  could you review it?

File: src/utils.jl [hunk 1]
```diff
@@ -56,3 +56,132 @@ function insorted(item, collection)
...
+Partition `1:num_items` into atmost `required_partitions` number of contiguous partitions with 
+the objective of minimising the largest partition.
+The size of a partition is equal to the num of the weight of its elements.
+`weights[i] > 0`.
+
+### Performance
+Time: O(num_items+required_partitions)
+Requires only one iteration over `weight` but may not output the optimal partition.
+
+### Imlementation Notes
```
Comment (sbromberger) [inline src/utils.jl]: typo: Im_p_lementation

Comment (SohamTamba) [reply src/utils.jl]: Done

**Comment by sbromberger** [comment]
is this ready to merge @SohamTamba @somil55 ?

**Comment by somil55** [comment]
Yes!

**Comment by SohamTamba** [comment]
@sbromberger Yes!

**Comment by sbromberger** [comment]
Cool! Thanks!

**Pull request merged**